### PR TITLE
Remove the use of `Block` from `aead::Tag`.

### PR DESCRIPTION
**Pull request by unseddd**
Related to #1021

### CODE REVIEW

File: src/aead.rs [hunk 1]
```diff
@@ -635,11 +635,11 @@ impl Eq for Algorithm {}
 /// An authentication tag.
 #[must_use]
 #[repr(C)]
-pub struct Tag(Block);
+pub struct Tag([u8; BLOCK_LEN]);
```
Comment (briansmith) [inline src/aead.rs:638]: We have `TAG_LEN` that we should use instead.

File: src/aead.rs [hunk 2]
```diff
@@ -635,11 +635,11 @@ impl Eq for Algorithm {}
...
 
 impl AsRef<[u8]> for Tag {
     fn as_ref(&self) -> &[u8] {
-        self.0.as_ref()
+        &self.0[..]
```
Comment (briansmith) [inline src/aead.rs:642]: `self.0.as_ref()`?

File: src/aead/poly1305.rs [hunk 3]
```diff
@@ -160,7 +160,7 @@ impl Funcs {
 
     #[inline]
     fn emit(&self, state: &mut Opaque, nonce: &Nonce) -> Tag {
-        let mut tag = Tag(Block::zero());
+        let mut tag = Tag([0u8; BLOCK_LEN]);
```
Comment (briansmith) [inline src/aead/poly1305.rs:163]: Ditto: Use `TAG_LEN`.

**Comment by unseddd** [comment]
Let me know if you want me to rebase on current `main`.

**Comment by briansmith** [comment]
Thanks!

**Pull request merged**